src/rebuild/scan.c

```diff
@@ -862,6 +862,13 @@ rebuild_container_scan_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 			return rc;
 		}
 		cont_child->sc_discarding = 1;
+		while (cont_child->sc_ec_agg_active) {
```

Comment:
so in agg_rate_ctl() check "cont->sc_discarding" to abort aggregation when it is true, it is not enough because agg_rate_ctl() only executed in yield_func()?
if need to exclude EC agg case, why need not to exclude  VOS agg "cont_chhild->sc_vos_agg_active" case?

but if exclude both EC agg and VOS agg, the "cont->sc_discarding" is not needed any more?

Reply:
>so in agg_rate_ctl() check "cont->sc_discarding" to abort aggregation when it is true, it is not enough because agg_rate_ctl() only executed in yield_func()?

discarding is for aborting the aggregation. so ec_aggregate can exit the loop asap, then reintegrating should block the EC aggregation to enter the aggregation, and also set sc_ec_agg_active to 0.

> if need to exclude EC agg case, why need not to exclude VOS agg "cont_chhild->sc_vos_agg_active" case?

oh, aggregate_enter() should exclude the vos_aggregation and discard() for vos_aggregate. So we only need exclude ec_aggregation with discard here. I thought about exporting the aggregate_enter() for ec_aggregation, but it seems easier to exclude the aggregate here.